### ISSUE TITLE
Multiple issues [Trusty: Gnome] (Launching Anki, UTF-8 Locale, Commands not found, Terminal Font Size)

### ISSUE THREAD

**Issue by TsinghuaMike** [dnschneid/crouton#1894]
Hey guys,

I've got the gnome version of crouton set up. In particular, I took this command to set it up from a blog post, and just replaced "lxde" with "gnome" and got a much better set up this time.

sudo sh -e ~/Downloads/crouton -a i386 -r trusty -t audio,core,gtk-extra,keyboard,x11,chrome,cli-extra,extension,lxde

(I should say this is my 3rd go at it. Where the first time I installed gnome and got a bare-bones set up. I got frustrated and powerwashed and started over with xfce. I found I had the same exact problems with not running anki, and that's when I found this blog post that allows me to set up a proper gnome desktop)

Now that said... I have multiple issues, and I'm a Linux novice! One of these images is from the xfce version, which had a reasonable size terminal font. The next shows the font problem in my current set up. I tried this question about anki on the anki forum, and here's a response I got:

"I've seen this before, but I don't remember exactly what the fix was. This looks like it might be helpful:

http://ubuntuforums.org/showthread.php?t=2116564

This is also related, but it's for Arch Linux, so the commands might be different:

https://bbs.archlinux.org/viewtopic.php?id=155883"

And having checked out those links and trying those commands, I just get something like "command not found."

So I tried this, and a couple other things for the UTF-8 problem: 
locale-gen en_US
locale-gen en_US.UTF8

Another issue with this version, and it could be related to the small fonts, is that I can't copy and paste things into the terminal. Doesn't seem to do anything when I right click to paste.

So that's the problem in a nutshell. I'm trying my best, and I really appreciate the help. Crouton is an amazing idea, it's the reason I decided to take a chance and buy a Chrome Pixel 2015 and take a stab at learning Linux. Thanks for all the hard work you and everyone else put into this.

Edit: One last issue I've had is with the keyboard shortcuts working to switch back and force. If i'm in Linux, I can't seem to press ctrl+alt+back to go back to chrome. And the other command doesn't bring me to Linux.

![smallfont](https://cloud.githubusercontent.com/assets/11172428/8351741/8ca5d17a-1afd-11e5-96cf-251f7a973b21.png)

![img_20150624_145903](https://cloud.githubusercontent.com/assets/11172428/8351456/65583d94-1afb-11e5-8530-44281dd0f246.jpg)
![myscreenshot](https://cloud.githubusercontent.com/assets/11172428/8351459/6e5e3132-1afb-11e5-856e-ef60e84ce8f4.png)

**Comment by Zane-** [comment]
To change Xterm's DPI and fonts etc, you need to create a file called ".Xresources" in your home directory. After any changes you need to enter "xrdb -merge ~/.Xresources" then restart the terminal. Here is my .Xresources file. I think it looks quite nice. (It is set up for the Chromebook Pixel)
XTerm*faceName: Bitstream Vera Serif Mono
xterm*faceSize: 11
xterm*vt100*geometry: 80x22
xterm*saveLines: 16384
xterm*loginShell: true
xterm*charClass: 33:48,35:48,37:48,43:48,45-47:48,64:48,95:48,126:48
xterm*termName: xterm-color
xterm*eightBitInput: false
Xft.dpi: 190

!BLK Cursor
#define _color0        #000d18
#define _color8        #000d18
!RED Tag
#define _color1        #e89393
#define _color9        #e89393
!GRN SpecialKey
#define _color2        #9ece9e
#define _color10       #9ece9e
!YEL Keyword
#define _color3        #f0dfaf
#define _color11       #f0dfaf
!BLU Number
#define _color4        #8cd0d3
#define _color12       #8cd0d3
!MAG Precondit
#define _color5        #c0bed1
#define _color13       #c0bed1
!CYN Float
#define _color6        #dfaf8f
#define _color14       #dfaf8f
!WHT Search
#define _color7        #efefef
#define _color15       #efefef
!FMT Include, StatusLine, ErrorMsg
#define _colorBD       #ffcfaf
#define _colorUL       #ccdc90
#define _colorIT       #80d4aa
!TXT Normal, Normal, Cursor
#define _foreground    #dcdccc
#define _background    #1f1f1f
#define _cursorColor   #8faf9f
URxvt*color0         : _color0
URxvt*color1         : _color1
URxvt*color2         : _color2
URxvt*color3         : _color3
URxvt*color4         : _color4
URxvt*color5         : _color5
URxvt*color6         : _color6
URxvt*color7         : _color7
URxvt*color8         : _color8
URxvt*color9         : _color9
URxvt*color10        : _color10
URxvt*color11        : _color11
URxvt*color12        : _color12
URxvt*color13        : _color13
URxvt*color14        : _color14
URxvt*color15        : _color15
URxvt*colorBD        : _colorBD
URxvt*colorIT        : _colorIT
URxvt*colorUL        : _colorUL
URxvt*foreground     : _foreground
URxvt*background     : _background
URxvt*cursorColor    : _cursorColor
XTerm*color0         : _color0
XTerm*color1         : _color1
XTerm*color2         : _color2
XTerm*color3         : _color3
XTerm*color4         : _color4
XTerm*color5         : _color5
XTerm*color6         : _color6
XTerm*color7         : _color7
XTerm*color8         : _color8
XTerm*color9         : _color9
XTerm*color10        : _color10
XTerm*color11        : _color11
XTerm*color12        : _color12
XTerm*color13        : _color13
XTerm*color14        : _color14
XTerm*color15        : _color15
XTerm*colorBD        : _colorBD
XTerm*colorIT        : _colorIT
XTerm*colorUL        : _colorUL
XTerm*foreground     : _foreground
XTerm*background     : _background
XTerm*cursorColor    : _cursorColor

**Comment by TsinghuaMike** [comment]
Hey Zane,

I really appreciate your thoughtful reply. But, man, is this getting frustrating. I've followed all kinds of instructions regarding terminal commands, and nothing would help me install any tar files. (Is it normal to only have access to an X and UXTerm but not a regular "Terminal"?). The only things I was able to install were .deb files, and even most of those wouldn't run for various reasons. My gnome installation didn't even have a text editor, and it wouldn't let me install one, so I couldn't create an .xresources file. I've had all I could take trying to install basic things, so I decided to powerwash the system and try again. I figure the best way to solve this problem is to find an installation that has all the basic things pre-loaded in the system. Does anyone have an improvement on this command: 
sudo sh -e ~/Downloads/crouton -a i386 -r trusty -t audio,core,gtk-extra,keyboard,x11,chrome,cli-extra,extension,gnome

Basically I want the installation to include as much as possible to help a Linux novice get started. I would hope whatever I install would allow me to utilize the volume keys on the keyboard, for example. And if it has to be unity, then I'm cool with that, too.

I have a pixel, so the shift-control-alt command doesn't work. My ctrl-alt-right ctrl-alt-refresh command just causes me problems. It sends me to a terminal looking screen, and I can only get out of it by pressing esc-refresh-power. When I run Linux, ctrl-alt-back doesn't work.

Anyway, I'm patient. And I really love the speed and design of this computer. So I'm just toughing it out. And I appreciate the support of the open source community. You guys are the greatest.

--Mike

**Comment by Zane-** [comment]
Hey Mike, 

Once again, Ctrl-Alt-Back is depreciated, use Ctrl-Alt-Shift-Back to switch from Linux to Chrome-OS.

You don't install tar files, they are an archive file (tape archive). Typically you would untar something using the command: "tar xvf file.tar". Source code is commonly packaged in tar files so perhaps that's what you meant by install. 

A basic text editor you can install is "gedit". You can install it by running "sudo apt-get install gedit" in your terminal. You can then run it by using "gedit filename".

There aren't really any crouton targets you can add that will make your linux environment novice-friendly.

I will write what I do, as I have the same device as you and install the same DE and distro. Please note I am typing all this from memory, so forgive me if I get a name wrong.

To get a trusty gnome chroot going: 
sudo sh -e ~/Downloads/crouton -r trusty -t gnome,extension

The extension target will allow the clipboard to be synchronized between chrome-os and linux. You will have to install the crouton integration extension on the Chrome Web Store.

After my chroot is complete installing, I enter it by entering "sudo startgnome"

The first thing I do is get the terminal looking nice, as you will be using it a lot. Open Xterm and type:
sudo apt-get install gedit
After it is finished installing, make sure your working directory is your home directory by typing:
cd
This will change the directory to your home directory.
Then type:
gedit .Xresources
This should create a blank file named .Xresources and open it in the editor.
You can then just copy and paste my code in my other reply then in Xterm enter:
xrdb -merge ~/.Xresources
Close Xterm and restart it, it should be bigger and have a nice color scheme.

Now we want to install some basic packages.
Let's get Ubuntu Software Center installed, in Xterm type:
sudo apt-get install software-center
Just hit Y when it asks if you want to install something
Note: If something ever fails to install and you see dependency error, you can force it correct those errors by typing "sudo apt-get install -f"

After the software center is done installing, let's get synaptic package manager. Type this in:
sudo apt-get install synaptic
After synaptic is finished installing let's go ahead and open that to install pulseaudio. This will make sound work in linux. After synaptic is open search "pulseaudio", right click pulseaudio, click "Mark for installation", then search firefox, mark firefox for installation, then install the selected packages.

Now to fix the small mouse cursor. Type this into Xterm:
sudo apt-get install dconf-editor
After that is finished, type dconf-editor into Xterm. This should open the dconf-editor program.
Now you should see a menu on the left, follow this path:
org > gnome > desktop > interface
There you should see cursor-size. I suggest you change the value to 48.
You can also mess around with the other various settings there.

Now let's fix the UI of firefox. Go ahead and start firefox. Everything should be tiny. To fix this type this into the URL, about:config. Just hit I understand to the warning, then type this into the search bar: 
layout.css.Dev
You should see Devpixelsperpx, change this to 2.
Now to fix the blurry icons, google "Gnome 3 firefox theme", then install that.

We should now set up some keyboard shortcuts to make our lives easier.
Search "keyboard" on the applications screen and open the Keyboard settings panel. Navigate to the shortcuts tab, then go to custom shortcuts.

Here are some that I add:
Name: Logout
Command: gnome-session-quit
I bind this one to Super (The search key) + L
This makes it quick and easy to logout of linux.
Name: Launch Xterm
Command: xterm
I bind this one to Super + T
Name: Launch Firefox
Command: firefox
I bind this one to Super + W
Name: Launch File Manager
Command: nautilus
I bind this one to Super + S

I would recommend looking through the various default keybinds you can assign. They can really speed up your workflow.

If you set that Logout keyboard shortcut, you can logout by pressing Search + L then pressing enter. 

Every time I do something major to my chroot I make a backup to save my progress. You can do this by typing this into the chrome-os shell:
sudo edit-chroot -b trusty
I usually put these on google drive or a USB flash drive in case something goes wrong.
You can restore backups using:
sudo edit-chroot -r trusty
This will work if you botch something up and need to restore to a previous state.
If you powerwashed and you want to install your backup use this command:
sudo sh ~/Downloads/crouton -f "backupfile"

Let me know if you have any additional questions or concerns or if something goes wrong. I left a lot out.

-Zane

**Comment by TsinghuaMike** [comment]
Man, Zane, I can't even tell you how awesome you are for that message. I'm fighting off jetlag as we speak and I'm going to get to work on this as soon as I get a nap in. I offer you an internet high-five

--Mike

**Comment by TsinghuaMike** [comment]
Just as expected, I am up at 4:00am with jet-lag.

So far we have some successes and failures. The first success is this time my installation is a 64 bit and not 32 bit. It seems like the better choice. I was able to install gedit fairly easily, and create this .Xresources file fairly easily. I had no problem installing firefox, although it seems a bit strange not to have a resizing window. I can live with that. 

I typed "cd" like you said, and it seems to make a difference. Things are going in the home folder. I have the Xresources file just like you suggested. 

And I typed in:

![terminaltext](https://cloud.githubusercontent.com/assets/11172428/8373641/98787664-1bbc-11e5-92c2-dd0a9fe0bd07.png)

I'll just take a minute here and pat myself on the back for using ImageMagick, but my terminal window is still micro sized.

![xresources](https://cloud.githubusercontent.com/assets/11172428/8373508/93e00bc2-1bbb-11e5-97cd-dd80af8f5856.png)

The crouton integration chrome app hasn't kicked in yet. I was able to get it to work before, not sure what I did differently. Running it the way I am now, I have noticed that sound doesn't work. I believe in my previous installation my command added sound functionality. And although it seems that my locale is listed as en_UTF.US-8 or something like that, which is supposed to be the one without problems, I still haven't been able to run my favorite app Anki. 

Gnome extensions and firefox extensions are great. I think I can really enjoy using Gnome. Looks really quite nice in my opinion.

ctrl+alt+shift+back is still doing nothing. That's the least my problems at the moment, though. I was able to install the ubuntu software center and synaptic without any problem. So that's an update.

--Mike

**Comment by DennisLfromGA** [comment]
@Zane- 

Thanx for taking the time and effort in answering @TsinghuaMike questions and explaining your process and setup. This is very helpful and, even though I don't (yet) have a Pixel, the info. is very enlightening and useful for me as well. 

I've gleaned a lot of great info. from your replies. You may want to consider a wiki page or two on this site so that others can reap the benefits of your experience.

Thanx again,
-DennisL

**Comment by Zane-** [comment]
Mike, 
You have to close the terminal and open it again for the changes to take effect.
You can get minimize and maximize buttons on your windows by installing gnome tweak tool
Install it using:
`sudo apt-get install gnome-tweak-tool`
Open that up and go to the Windows sections. You should see an option that will enable Maximize and Minimize buttons.

As for sound, did you install pulseaudio via synaptic? That's what gets sound going.

**Comment by TsinghuaMike** [comment]
Hey again Zane (and others),

I have to say I got everything set up -- thanks to you -- and I'm really starting to get the hang of linux. All of those programs you recommended are great. And I even found some more from the ubuntu software center.

Except now I came across this unexpected problem. When I launch through crouton integration, it gives me this error: 

![screenshot 2015-06-27 at 12 22 21 pm](https://cloud.githubusercontent.com/assets/11172428/8392934/1eb109ec-1cc8-11e5-89d2-f968d522649f.png)

The first time getting this, I was able to still get in after some time, and a little luck, but now I'm out of luck.

I think I understand the problem. With crouton integration I have a general issue with screen resolution. For example, my firefox browser is huge and not enjoyable to use. I've messed around with the screen resolution settings a bunch and even after selecting the smallest setting everything is larger than life. I tried a few times to launch gnome the normal way [In other words - not through a window on chrome os], and it just won't launch. Maybe I can fix this issue if I can get back into it that way?

--Mike

**Comment by Zane-** [comment]
Mike, 

I don't know much about running crouton in a window (xiwi). I have never used it as I want hardware acceleration. If your chroot won't start running `sudo startgnome`, you could try updating it. That's worked for me in the past. To update your chroot use the command:
`sudo sh crouton -r trusty -t gnome -u`
After the update is complete, try starting it again via `sudo startgnome`
If that didn't work, then I wouldn't know how to fix it. This is why backups are so important. Hopefully you made one after setting things up.

-Zane

**Comment by TsinghuaMike** [comment]
Yeah looks like I'll have to start over. Not a big deal, because I know what I'm doing this time. And I'll set up a backup before I use the chrome crouton extension.

**Comment by TsinghuaMike** [comment]
Hey Zane,

I have one main issue to take care of before I backup my file:

I've seen this possibly reported before here: https://github.com/dnschneid/crouton/issues/1519

But i'm not sure if this is the same problem. So I'm posting my workspace. The reason I think it's different is because I typed in this exact command:

sudo apt-get install software-properties-common python-software-properties
sudo add-apt-repository https://download.01.org/gfx/ubuntu/14.04/main
wget --no-check-certificate https://download.01.org/gfx/RPM-GPG-KEY-ilg -O - | sudo apt-key add -
wget --no-check-certificate https://download.01.org/gfx/RPM-GPG-KEY-ilg-2 -O - | sudo apt-key add -
sudo apt-get update
sudo apt-get upgrade

Nothing is different about my screen. And it actually looks great except for words getting distorted.

Anyway, that's the main problem, and these other ones are not really that important in comparison, so no worries if you don't know how to help.

My other problem I don't think is related to crouton but to linux, and that's setting up a program called netlogo. I installed java, so I'm not exactly sure why it doesn't run. I think I'm supposed to just click on the file "netlogo.sh" and "run software", but nonetheless it just gives me a waiting signal and crashes.

Oh, and  did also have trouble with the keyboard shortcuts. I set up the volume keys to work on super+volume, but it didn't do anything. Maybe I missed a step?

![workspace 1_001](https://cloud.githubusercontent.com/assets/11172428/8397965/1fe9dce6-1dac-11e5-97e0-c82f927337e3.png)
![netlogo](https://cloud.githubusercontent.com/assets/11172428/8397966/28cc799a-1dac-11e5-95c4-3ed2b11397d5.png)

**Comment by Zane-** [comment]
Mike, 

Installing the graphics driver doesn't really change how your OS looks. It improves the performance of games, etc. fixes graphical glitches that occur when you start the chroot and when you switch to Chrome OS.

As for the text rendering bug, I've had that too. That occurs when you let the Ubuntu Software Updater update the driver (After you update it via apt-get). You have to deselect the graphics driver when it says new software is available. If you already let it update you're screwed, or at least I think you are.

Honestly it's a lot of tinkering around to get things to work properly. For the java not working, the fix was copying over a desktop configuration file. I'd rather not revisit that as it took my quite a while digging through bug reports to find the proper file.

The volume shortcuts probably don't work because you didn't install pulseaudio. If you included the sound target when you initially set up your chroot, then I don't know, it might take priority over pulseaudio or something. I only do `-t gnome,extension`.

If you'd like I can share my backup. It has pretty much everything fixed. Graphics + touchpad drivers, sound and brightness shortcuts, Java fixed, etc.

-Zane

**Comment by TsinghuaMike** [comment]
Yeah, maybe that's the best solution, to use your backup rather than go through the whole process which takes a few hours. How do you share that?

**Comment by Zane-** [comment]
I would just upload it somewhere then share the link. I've prepared a backup file with all my accounts logged out and everything. My upload speed here is really bad though so I'll upload it to Google Drive next time I go to Starbucks and share the link.

**Comment by TsinghuaMike** [comment]
Sounds great. I can't thank you enough for helping me transition to crouton

--Mike

**Comment by TsinghuaMike** [comment]
Hey Zane,

No need to send me that file.
I did everything over, and made a backup, so now I'm just trying to figure out how to restore it:

![screenshot 2015-07-05 at 1 13 14 pm](https://cloud.githubusercontent.com/assets/11172428/8512579/af889714-2317-11e5-8a0e-c6cabb67b3db.png)

I tried those commands and they didn't work, so I figure I might have to do something with the file itself:

![screenshot 2015-07-05 at 1 14 49 pm](https://cloud.githubusercontent.com/assets/11172428/8512582/de43b624-2317-11e5-81f8-eb22a0897e6b.png)

It's in a tar.gz format. I don't know how to change that format. Google suggested to use this app called wakame but it always crashes. Do I have to alter it in another computer and then load it from a flashdrive?

**Comment by Zane-** [comment]
Restoring depends on if you're restoring an exisiting chroot (you have used crouton before) or if you powerwashed. If you powerwashed, download crouton and do:
(Quoted from the Wiki https://github.com/dnschneid/crouton/wiki):

"In the event that you want to restore a backup to a stock chromebook (with developer mode enabled) you can use the crouton installer with the -f option specifying the backup's tarball, e.g., sudo sh -e crouton -r raring -t xfce -f /media/removable/SD\ Card/backup/raring-20130617-1234.tar.gz"

So your command would probably be:
`sudo sh -e crouton -r trusty -t gnome -f ~/Downloads/trusty*.tar.gz`
